test/app_test.rb

```diff
@@ -3,14 +3,27 @@
 require_relative '../app'
 require 'minitest/autorun'
 require 'rack/test'
+require 'pg'
 
 ENV['RACK_ENV'] = 'test'
 
 class AppTest < MiniTest::Test
   include Rack::Test::Methods
 
+  MOCK_TEST = 0
+  JSON_TEST = 1
+  PGDB_TEST = 2
+
   def app
-    initialize_memo_file
+    # switch test
+    @test_number = PGDB_TEST
+
+    case @test_number
+    when JSON_TEST
+      initialize_memo_file
+    when PGDB_TEST
+      initialize_memo_db
+    end
```

Comment:
```suggestion
    @test_type = :pgdb

    case @test_type
    when :json 
      initialize_memo_file
    when :pgdb
      initialize_memo_db
    end
```

数値で種別を管理するより、ハッシュを使った方がRubyらしくてなおかつ人間に優しいかも？と思いました。